File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.js
function solution(numbers) {
    return numbers.map(number => {
        let num = BigInt(number);

        // 짝수인 경우
        if (num % 2n === 0n) {
            return Number(num + 1n);
        }

        const rightmostZero = (num + 1n) & (~num);
        const nextOne = rightmostZero >> 1n;

        return Number(num ^ rightmostZero ^ nextOne);
    });
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];

        for (int i = 0; i < numbers.length; i++) {
            long num = numbers[i];

            // 짝수인 경우
            if (num % 2 == 0) {
                answer[i] = num + 1;
                continue;
            }

            long rightmostZero = (num + 1) & (~num);
            long nextOne = rightmostZero >> 1;

            answer[i] = num ^ rightmostZero ^ nextOne;
        }

        return answer;
    }
}